Add the relevant g++ compiler package to the docker images

mattface opened the pull request:
Ard has requested g++ host compiler be installed on the builder for GCC
plugins.

khilman commented:
Looks like a package that belongs in base, and not duplicated in all the arch-specific Dockerfiles?

gctucker commented:
Indeed.

mattface commented:
Note that it's a different version between gcc7 and gcc8, as discussed in the meeting, so it can't be in base.

gctucker commented:
I don't think the toolchain used for building user-space tools matters.  At least it shouldn't be related to the toolchain version used to build the kernel.

gctucker commented:
Also, we already have the default `gcc` host compiler in the build-base image, following the same principle.

If we really want to deal with different versions of the host tools, then imo we should have another set of intermediate Dockerfiles i.e. build-base-gcc-7 and build-base-gcc-8.  But I don't think that's necessary.

mattface commented:
Why is host gcc installed in build-base? It was not in my initial docker images, it wasn't needed.

I asked in the meeting if the version mattered and I was under the belief that it was agreed, if one of these user space builds fails then it's confusing to see the build job failed with "gcc-8" where actually debian installed gcc-7 as a default host compiler.

Having intermediary images means that you end up with a dependency chain you have to monitor yourself, as docker doesn't rebuild the chain if you change something in the middle or top. It's why my initial images didn't have a build-base.

gctucker commented:
Host `gcc` was always there, it's a dependency from some other package.  You can't cross-compile the kernel without a host compiler to produce some user-space build tools (e.g. `dtc`...).

I've rebuilt your original `Dockerfile.gcc-7_arm64` image, here's what `apt` is telling us:
```
gtucker@submarine:~/project/kernelci/kernelci-core/jenkins/dockerfiles [(HEAD detached at f62ece4884db)]
$ docker build . -f Dockerfile.gcc-7_arm64 -t gtucker/kernelci-build-old-gcc-7_arm64
Sending build context to Docker daemon  17.92kB
Step 1/11 : FROM debian:buster
 ---> 7edbed26ed12
Step 2/11 : MAINTAINER "Matt Hart" <matt@mattface.org>
 ---> Using cache
 ---> c5c8fc92bcc1
Step 3/11 : RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get -y install bc bsdmainutils make bison flex ccache wget libelf-dev tar python2.7 python-requests python-pyelftools libssl-dev lzop u-boot-tools kmod
 ---> Running in 4358b20b378d
Get:1 http://security-cdn.debian.org/debian-security buster/updates InRelease [38.3 kB]
Get:2 http://cdn-fastly.deb.debian.org/debian buster InRelease [150 kB]
Get:3 http://cdn-fastly.deb.debian.org/debian buster-updates InRelease [47.6 kB]
Get:4 http://cdn-fastly.deb.debian.org/debian buster/main amd64 Packages [7727 kB]
Fetched 7963 kB in 2s (3460 kB/s)
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
tar is already the newest version (1.30+dfsg-2).
tar set to manually installed.
The following additional packages will be installed:
  binutils binutils-common binutils-x86-64-linux-gnu bzip2 ca-certificates cpp
  cpp-8 device-tree-compiler file gcc gcc-8 gcc-8-base libasan5 libatomic1
  libbinutils libbison-dev libbsd0 libc-dev-bin libc6 libc6-dev libcc1-0
  libexpat1 libfl-dev libfl2 libgcc-8-dev libgcc1 libgomp1 libgpm2 libisl19
  libitm1 libkmod2 liblsan0 liblzo2-2 libmagic-mgc libmagic1 libmpc3 libmpfr6
  libmpx2 libncurses6 libncursesw6 libpsl5 libpython-stdlib libpython2-stdlib
  libpython2.7-minimal libpython2.7-stdlib libquadmath0 libreadline7
  libsigsegv2 libsqlite3-0 libssl1.1 libstdc++6 libtinfo6 libtsan0 libubsan1
  linux-libc-dev lsb-base m4 manpages manpages-dev mime-support openssl
  publicsuffix python python-asn1crypto python-certifi python-cffi-backend
  python-chardet python-cryptography python-enum34 python-idna
  python-ipaddress python-minimal python-openssl python-pkg-resources
  python-six python-urllib3 python2 python2-minimal python2.7-minimal
  readline-common xz-utils zlib1g-dev
Suggested packages:
  binutils-doc bison-doc wamerican | wordlist whois vacation bzip2-doc distcc
  | icecc cpp-doc gcc-8-locales build-essential flex-doc gcc-multilib autoconf
  automake libtool gdb gcc-doc gcc-8-multilib gcc-8-doc libgcc1-dbg
  libgomp1-dbg libitm1-dbg libatomic1-dbg libasan5-dbg liblsan0-dbg
  libtsan0-dbg libubsan1-dbg libmpx2-dbg libquadmath0-dbg glibc-doc libc-l10n
  locales gpm libssl-doc m4-doc make-doc man-browser python-doc python-tk
  python-cryptography-doc python-cryptography-vectors python-enum34-doc
  python-openssl-doc python-openssl-dbg python-setuptools python-socks
  python-ntlm python2-doc python2.7-doc binfmt-support readline-doc
The following NEW packages will be installed:
  bc binutils binutils-common binutils-x86-64-linux-gnu bison bsdmainutils
  bzip2 ca-certificates ccache cpp cpp-8 device-tree-compiler file flex gcc
  gcc-8 kmod libasan5 libatomic1 libbinutils libbison-dev libbsd0 libc-dev-bin
  libc6-dev libcc1-0 libelf-dev libexpat1 libfl-dev libfl2 libgcc-8-dev
  libgomp1 libgpm2 libisl19 libitm1 libkmod2 liblsan0 liblzo2-2 libmagic-mgc
  libmagic1 libmpc3 libmpfr6 libmpx2 libncurses6 libpsl5 libpython-stdlib
  libpython2-stdlib libpython2.7-minimal libpython2.7-stdlib libquadmath0
  libreadline7 libsigsegv2 libsqlite3-0 libssl-dev libssl1.1 libtsan0
  libubsan1 linux-libc-dev lsb-base lzop m4 make manpages manpages-dev
  mime-support openssl publicsuffix python python-asn1crypto python-certifi
  python-cffi-backend python-chardet python-cryptography python-enum34
  python-idna python-ipaddress python-minimal python-openssl
  python-pkg-resources python-pyelftools python-requests python-six
  python-urllib3 python2 python2-minimal python2.7 python2.7-minimal
  readline-common u-boot-tools wget xz-utils zlib1g-dev
The following packages will be upgraded:
  gcc-8-base libc6 libgcc1 libncursesw6 libstdc++6 libtinfo6
6 upgraded, 91 newly installed, 0 to remove and 18 not upgraded.
Need to get 52.6 MB of archives.
```
As you can see, `gcc-8` was there.  We can simply add `g++` to the base image, and check with Ard if that's OK for what he needs.

ana commented on jenkins/dockerfiles/gcc-8_arm/Dockerfile:
```diff
@@ -1,7 +1,7 @@
 
 RUN apt-get update && apt-get install --no-install-recommends -y \
-    gcc-8-arm-linux-gnueabihf
+    gcc-8-arm-linux-gnueabihf g++-8
```
should this be `g++-8-arm-linux-gnueabihf` ?

ana replied:
Nah, I just saw the exchange above.

mattface commented:
Moved all the g++ packages back to build-base, it will install whatever you get from debian, should be good enough.
Also rebased to include the xz utils patch

gctucker commented:
Cool.  I guess the 2 commits can just be squashed?
